puppet/chassis.rb

```diff
@@ -229,4 +230,103 @@ def self.make_relative(base, paths)
 			pathname.relative_path_from(base).to_s
 		end
 	end
+
+	def self.update_core
+		puts "\e[32mChecking for Chassis core updates...\e[0m"
+		core = ['core']
+		updates = self.updates_check(core, @@dir)
+		self.prompt_for_updates(updates, @@dir, 'core')
+	end
+
+	def self.update_submodules
+		puts "\e[32mChecking for Chassis submodule updates...\e[0m"
+		submodules = ['apt','mysql','stdlib','wp']
+		directory = File.join(@@dir, 'puppet/modules')
+		updates = self.submodule_update_check(submodules, directory)
+	end
+
+	def self.update_extensions
+		puts "\e[32mChecking for Chassis extension updates...\e[0m"
+		extensions = Dir.glob(@@extension_dir + '/*').map { |directory| File.basename( directory ) }
+		updates = self.updates_check(extensions, @@extension_dir)
+		self.prompt_for_updates(updates, @@extension_dir, 'extensions')
+	end
+
+	def self.update_global_extensions
+		puts "\e[32mChecking for Chassis global extension updates...\e[0m"
+		global_ext_path = File.join(Dir.home, ".chassis", "extensions")
+		global_extensions = Dir.glob(global_ext_path + '/*').map { |directory| File.basename( directory ) }
+		updates = self.updates_check(global_extensions, global_ext_path)
+		self.prompt_for_updates(updates, global_ext_path, 'extensions')
+	end
+
+	def self.updates_check(folders, directory)
+		updates = Array.new
+		folders.each do |folder|
+			next if folder == 'example'
+			if 'core' == folder
+				folder = directory
+				Dir.chdir(directory)
+			elsif
+				Dir.chdir(directory + '/' + folder )
+			end
+			git_checkout_master_stdout, git_checkout_master_stdeerr, git_checkout_master_status = Open3.capture3("git checkout master")
+			git_remote_update_stdout, git_remote_update_stdeerr, git_remote_update_status = Open3.capture3("git remote update")
+			git_status_stdout, git_status_stdeerr, git_status_status = Open3.capture3("git status --porcelain=2 --branch")
```

Comment:
Since we're not using most of these variables, can we just call the function and ignore the return? We should probably check the status, so maybe:
```ruby
_, err, status = Open3...
raise err if status != 0

_, err, status = Open3...
```

(`_` means an unused var in Ruby; it's handled differently by the language IIRC)

Reply:
TIL `_` 🕺